s3transfer/futures.py

```diff
@@ -219,14 +224,30 @@ def cancel(self):
         """Cancels the TransferFuture"""
         if not self.done():
             with self._lock:
```

Comment:
I can add the lock to now before ``self.done()``. I removed the lock around status and so before this would cause a deadlock.